buildkitd/docker-auto-install.sh

```diff
@@ -110,10 +118,33 @@ install_dockerd_amazon() {
     esac
 }
 
+install_jq() {
+    case "$distro" in
+        alpine)
+            apk add --update --no-cache jq
+            ;;
+
+        amzn)
+            yum install jq
+            ;;
+
+        *)
+            export DEBIAN_FRONTEND=noninteractive
+            apt-get update
```

Comment:
Does this cause a duplicate `apt-get update`? One for dockerd and one for jq?

Reply:
that's a good point.

How does a test like

```
if [ -f /var/lib/apt/periodic/update-success-stamp ] && [ -z "$(find /var/lib/apt/periodic/update-success-stamp -mmin +50)" ]; then
    echo no update is needed
fi
```

look for detecting when we can skip it.

Reply:
Yeah, that could work - another idea could be to leave a file around the first time we run apt, and skip if we see the file.

Reply:
I was able to use a variable in the sh script to do this.